### PR TITLE
Update SellCommand.cs

### PR DESCRIPTION
**Pull request by xAstroBoy**
Make it sell by determined amount, filter off between scraps, defense weapons and held items as it steals off other's people held objects.
and make it as well include when the company is buying it as discounted rate when selling amount of straps for credits.

### CODE REVIEW

File: lc-hax/Scripts/Commands/SellCommand.cs [hunk 1]
```diff
@@ -16,12 +19,68 @@ public class SellCommand : ICommand {
...
+                    remainingAmount -= (int)discountedScrapValue;
+                }
+
+                scraps.RemoveAt(randomIndex);
+            }
+
+            Chat.Print(remainingAmount <= 0
+                ? $"Sold scraps to reach or exceed the target amount of {value}!"
+                : $"Could not sell enough scraps to reach the target amount. Remaining scrap: {remainingAmount}");
+        }
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Commands/SellCommand.cs:77]: Isn't this a pretty naive implementation? This problem can be solved by solving the [Knapsack Problem](https://en.m.wikipedia.org/wiki/Knapsack_problem).

File: lc-hax/Scripts/Commands/SellCommand.cs [hunk 2]
```diff
@@ -16,12 +19,68 @@ public class SellCommand : ICommand {
...
+
 
-            player.currentlyHeldObjectServer = grabbableObject;
+        void SellObject(GrabbableObject item) {
+            if (item == null || player == null || depositItemsDesk == null) return;
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Commands/SellCommand.cs:81]: Why is this null check necessary when you assigned `item` as a non-nullable type?

File: lc-hax/Scripts/Commands/SellCommand.cs [hunk 3]
```diff
@@ -16,12 +19,68 @@ public class SellCommand : ICommand {
...
+        if (args.Length is 1 && int.TryParse(args[0], out int targetAmount)) {
+            SellScrapAmount(targetAmount);
+            return;
+        }
+
+        void SellEverything() {
+            HaxObjects.Instance?.GrabbableObjects.ForEach(nullableGrabbableObject => {
+                if (!nullableGrabbableObject.IsNotNull(out GrabbableObject grabbableObject)) return;
+                if (!grabbableObject.itemProperties.isScrap) return;
+                if (grabbableObject.itemProperties.isDefensiveWeapon) return;
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Commands/SellCommand.cs:36]: Are some defensive items assigned as scrap?

**Review by winstxnhdw — CHANGES_REQUESTED** [review]

File: lc-hax/Scripts/Commands/SellCommand.cs (hunk 1 above)
Comment (xAstroBoy) [reply lc-hax/Scripts/Commands/SellCommand.cs:77]: Uh i did it based off my ways but if you can rewrite the code for it and  optimize it even further go ahead

File: lc-hax/Scripts/Commands/SellCommand.cs (hunk 2 above)
Comment (xAstroBoy) [reply lc-hax/Scripts/Commands/SellCommand.cs:81]: Just a habit i guess

File: lc-hax/Scripts/Commands/SellCommand.cs (hunk 3 above)
Comment (xAstroBoy) [reply lc-hax/Scripts/Commands/SellCommand.cs:36]: Yeah found out the shotgun is marked as scrap as well

File: lc-hax/Scripts/Commands/SellCommand.cs (hunk 1 above)
Comment (winstxnhdw) [reply lc-hax/Scripts/Commands/SellCommand.cs:77]: This isn't an optimisation per say. Your greedy algorithm makes it possible to occasionally miss the correct solution.

Comment (xAstroBoy) [reply lc-hax/Scripts/Commands/SellCommand.cs:77]: > This isn't an optimisation per say. Your greedy algorithm makes it possible to occasionally miss the correct solution.

i dont know how to resolve that issue, feel welcome to rewrite it using your best approach to it.

**Pull request merged**